File: src/Blazor.Toastr/Interop.cs
using Blazor.Toastr.Enums;
using Microsoft.JSInterop;
using Newtonsoft.Json;

namespace Blazor.Toastr
{
    internal static class Interop
    {
        public static ToastrOptionsInternal toastrOptions;

        internal static IJSRuntime JsRuntime { get; set; }

        public static void ShowToastr(ToastrType toastrType, string message, string title = null, ToastrOptions options = null)
        {
            switch (toastrType)
            {
                case ToastrType.Success:
                    JsRuntime.InvokeAsync<string>("toastrInterop.success", message, title, JsonConvert.SerializeObject(toastrOptions ?? options.ToInternal()));
                    break;
                case ToastrType.Info:
                    JsRuntime.InvokeAsync<string>("toastrInterop.info", message, title, JsonConvert.SerializeObject(toastrOptions ?? options.ToInternal()));
                    break;
                case ToastrType.Warning:
                    JsRuntime.InvokeAsync<string>("toastrInterop.warning", message, title, JsonConvert.SerializeObject(toastrOptions ?? options.ToInternal()));
                    break;
                case ToastrType.Error:
                    JsRuntime.InvokeAsync<string>("toastrInterop.error", message, title, JsonConvert.SerializeObject(toastrOptions ?? options.ToInternal()));
                    break;
                default:
                    break;
            }
        }
    }
}

File: src/Blazor.Toastr/content/Interop.js
window.toastrInterop = {
    success: function (message, title, options) {
        return toastr.success(message, title, $.parseJSON(options));
    },
    error: function (message, title, options) {
        return toastr.error(message, title, $.parseJSON(options));
    },
    info: function (message, title, options) {
        return toastr.info(message, title, $.parseJSON(options));
    },
    warning: function (message, title, options) {
        return toastr.warning(message, title, $.parseJSON(options));
    }
};
